### ISSUE TITLE
SPIR-V: Second level ternary op generating invalid SPIR-V

### ISSUE THREAD

**Issue by tafuri** [Microsoft/DirectXShaderCompiler#990]
Compiling the following hlsl shader generates invalid spir-v code: `fatal error: generated SPIR-V is invalid: Expected both objects to be of Result Type: Select`

```hlsl
uint TestF(int a, int b, int c)
{
    return a < b ? 0 : (a < c ? 1 : 2);
}
struct Input
{
    uint a : TEST0;
    uint b : TEST1;
    uint c : TEST2;
};
void main(in Input In, out uint Out : SV_TARGET)
{
    Out = TestF(In.a, In.b, In.c);
}
```

Chaing the `0` into `0u` in `TestF` fixes the problem.

**Comment by ehsannas** [comment]
Should be fixed by PR #995

**Comment by tafuri** [comment]
Great!
